src/main/java/edu/byu/cs/dataAccess/sql/helpers/SqlReader.java

```diff
@@ -0,0 +1,341 @@
+package edu.byu.cs.dataAccess.sql.helpers;
+
+import edu.byu.cs.dataAccess.DataAccessException;
+import edu.byu.cs.dataAccess.sql.SqlDb;
+import org.eclipse.jgit.annotations.NonNull;
+import org.eclipse.jgit.annotations.Nullable;
```

Comment:
After looking into this issue further, it seems that finding the right import is actually quite difficult. It seems there are plenty of people looking for an answer, but no one providing a standardized and maintained project for this.

See this [StackOverflow question](https://stackoverflow.com/questions/4963300/which-notnull-java-annotation-should-i-use) with 1300+ upvotes, a detailed synopsis of the lay of the land, but no satisfactory conclusion.

Based on the discussion in the thread above, it's clear that no standard answer exists. It seems the answer is mostly a matter of personal preference, and integration with whichever tools a particular project is depending on. Also, its seems pretty straightforward to add a maven dependency and search-and-replace a few `import` statements if we ever want to switch.

Since this project has no precedent, I'm going to assume that this approach is fine for now.